File: BE/lookey/src/main/java/com/project/lookey/path/dto/PlaceResponse.java
package com.project.lookey.path.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

@Data
@AllArgsConstructor
public class PlaceResponse {
    private int status;       // HTTP 상태 코드
    private String message;   // 응답 메시지

    private Result result;    // 실제 데이터

    @Data
    @AllArgsConstructor
    public static class Result {
        private List<Item> items;
    }

    @Data
    @AllArgsConstructor
    public static class Item {
        private String name;      // 편의점 이름
        private String address;   // 편의점 주소
        private Double lat;       // 위도
        private Double lng;       // 경도
        private Integer distance; // 현재 위치와의 거리(m)
        private String brand;     // GS25, CU 등

        @JsonProperty("place_id") // JSON 필드명을 place_id로 맞춤
        private String placeId;   // Kakao place id
    }
}

File: FE/ConvenienceSightApp/app/src/main/java/com/example/lookey/data/model/PlaceResponse.kt
package com.example.lookey.network.dto

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class PlaceResponseDto(
    val status: Int,
    val message: String,
    val result: ResultDto?
)

@JsonClass(generateAdapter = true)
data class ResultDto(
    val items: List<PlaceItemDto>
)

@JsonClass(generateAdapter = true)
data class PlaceItemDto(
    val name: String,
    val address: String,
    val lat: Double,
    val lng: Double,
    val distance: Int? = null,
    val brand: String? = null,
    val placeId: String? = null
)
